nmigen/back/rtlil.py

```diff
@@ -373,6 +370,9 @@ def on_Slice(self, value):
         if value.start == 0 and value.stop == len(value.value):
             return self(value.value)
 
+        if isinstance(value.value, ast.UserValue):
+            value.value = value.value._lazy_lower()
```

Comment:
You should never mutate any AST node anywhere in nMigen internals. That's one of the core concepts; you can elaborate/synthesize a design as many times as you want and it'll always work the same.